File: B_The_best_for_Geleta.java
import java.util.*;

public class Main{
    public static void main (String[] args){
        Scanner scanner = new Scanner(System.in);
        int rounds = scanner.nextInt();
        for (int t =0; t < rounds; t++){
            int n = scanner.nextInt();
            int m =scanner.nextInt();
            int [] arr = new int[n];
            for (int i=0; i<n; i++){
                arr[i] = scanner.nextInt();
            }
            for (int j=0;j<m;j++){
                String sign = scanner.next();
                int l = scanner.nextInt();
                int r = scanner.nextInt();

                for (int i=0;i<n;i++){
                    if (l <= arr[i] && arr[i]<=r){
                        arr[i] = (sign.equals("+"))? arr[i]+1:arr[i]-1

                    }
                }
                  System.out.print(Collections.max(Arrays.asList(Arrays.stream(arr).boxed().toArray((Integer[]::new))))+" ");
            }
          System.out.println();

        }

    }
}

File: B_The_best_for_Geleta.py
rounds = int(input())
for _ in range(rounds):
    n,m = list(map(int,input().split()))
    arr = list(map(int,input().split()))
    current_max = max(arr)
    answer =[]

    for _ in range(m):
      
        sign,l,r = list(input().split())
        l = int(l)
        r = int(r)
        for i in range(n):
            if l <= arr[i] <=r:
                arr[i] += 1 if sign =="+" else -1
        
        answer.append(max(arr))

    
    print(*answer)
   
            
        
       
    
~~